### PR TITLE
fix(panel): ensure close button is placed in the corner of the panel header.

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #6742

## Summary

Ensures the close button is always placed in the far corner of the panel header.

### CODE REVIEW

File: src/components/panel/panel.tsx [hunk 1]
```diff
@@ -387,12 +387,13 @@ export class Panel
...
 
   renderHeaderActionsEnd(): VNode {
-    const { close, hasEndActions, messages, closable } = this;
+    const { close, hasEndActions, messages, closable, hasMenuItems } = this;
     const text = messages.close;
 
     const closableNode = closable ? (
       <calcite-action
         aria-label={text}
+        data-test="close"
```
Comment (jcfranco) [inline src/components/panel/panel.tsx:396]: Can you create a follow-up issue to look into having these data attributes removed from production builds? I think we may be able to do that by using Stencil's `Build.isTesting` flag.

**Review by jcfranco — APPROVED** [review]
↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️
↗️❌↗️↗️↗️❌↗️❌❌❌↗️↗️❌❌❌↗️❌❌❌❌↗️❌↗️
↗️❌❌↗️↗️❌↗️↗️❌↗️↗️❌↗️↗️↗️↗️❌↗️↗️↗️↗️❌↗️
↗️❌↗️❌↗️❌↗️↗️❌↗️↗️❌↗️↗️↗️↗️❌❌❌↗️↗️❌↗️
↗️❌↗️↗️❌❌↗️↗️❌↗️↗️❌↗️↗️↗️↗️❌↗️↗️↗️↗️↗️↗️
↗️❌↗️↗️↗️❌↗️❌❌❌↗️↗️❌❌❌↗️❌❌❌❌↗️❌↗️
↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️↗️

File: src/components/panel/panel.tsx (hunk 1 above)
Comment (driskull) [reply src/components/panel/panel.tsx]: Can I just strip it out instead?

Comment (driskull) [reply src/components/panel/panel.tsx:396]: Might be a stencil bug, but I tried `data-test={Build.isTesting ? "close" : undefined}` and the test failed.

Comment (driskull) [reply src/components/panel/panel.tsx:396]: https://github.com/Esri/calcite-components/issues/6760

**Pull request merged**